src/util/taproot.rs

```diff
@@ -598,6 +598,15 @@ impl LeafInfo {
         }
     }
 
+    /// Creates an new [`LeafInfo`] with `script`, `ver` and merkle branch information.
+    pub fn with(script: Script, ver: LeafVersion, merkle_branch: TaprootMerkleBranch) -> Self {
```

Comment:
As mentioned previously, I don't like constructors here. I think we should let the internal algorithms do the proof construction. Plus, there is no API in rust-bitcoin that uses `LeafInfo` as an argument.